### ISSUE TITLE
[BUG] unable to sync wallet on testnet9

### ISSUE THREAD

**Issue by nextsux** [Chia-Network/chia-blockchain#6973]
**Describe the bug**
Wallet is unable to sync on testnet9:

```
2021-06-24T22:09:49.481 wallet chia.wallet.wallet_state_manager: INFO     Record for removed coin d35ea94d14a2553d67b4cbf588d114ead5fa0d4ca12d153887b36d01ce034fc2 is None. (ephemeral)
2021-06-24T22:09:49.482 wallet pool_wallet                : INFO     New PoolWallet singleton tip_coin: {'coin': {'amount': 1,
          'parent_coin_info': '0xba42277267f060672a51de6bf75ef3b94bcef71f5964421bfd99b28c3c0ef8e8',
          'puzzle_hash': '0x3960759df2397653c226d6a4a6a55591b3ab6b602178eebe77459f921dc2dd5f'},
 'puzzle_reveal': '0xff02ffff01ff02ffff01ff02ffff03ffff18ff2fffff010180ffff01ff02ff36ffff04ff02ffff04ff05ffff04ff17ffff04ffff02ff26ffff04ff02ffff04ff0bff80808080ffff04ff2fffff04ff0bffff04ff5fff808080808080808080ffff01ff088080ff0180ffff04ffff01ffffffff4602ff3304ffff0101ff02ffff02ffff03ff05ffff01ff02ff5cffff04ff02ffff04ff0dffff04ffff0bff2cffff0bff24ff3880ffff0bff2cffff0bff2cffff0bff24ff3480ff0980ffff0bff2cff0bffff0bff24ff8080808080ff8080808080ffff010b80ff0180ff02ffff03ff0bffff01ff02ff32ffff04ff02ffff04ff05ffff04ff0bffff04ff17ffff04ffff02ff2affff04ff02ffff04ffff02ffff03ffff09ff23ff2880ffff0181b3ff8080ff0180ff80808080ff80808080808080ffff01ff02ffff03ff17ff80ffff01ff088080ff018080ff0180ffffffff0bffff0bff17ffff02ff3affff04ff02ffff04ff09ffff04ff2fffff04ffff02ff26ffff04ff02ffff04ff05ff80808080ff808080808080ff5f80ff0bff81bf80ff02ffff03ffff20ffff22ff4fff178080ffff01ff02ff7effff04ff02ffff04ff6fffff04ffff04ffff02ffff03ff4fffff01ff04ff23ffff04ffff02ff3affff04ff02ffff04ff09ffff04ff53ffff04ffff02ff26ffff04ff02ffff04ff05ff80808080ff808080808080ffff04ff81b3ff80808080ffff011380ff0180ffff02ff7cffff04ff02ffff04ff05ffff04ff1bffff04ffff21ff4fff1780ff80808080808080ff8080808080ffff01ff088080ff0180ffff04ffff09ffff18ff05ffff010180ffff010180ffff09ff05ffff01818f8080ff0bff2cffff0bff24ff3080ffff0bff2cffff0bff2cffff0bff24ff3480ff0580ffff0bff2cffff02ff5cffff04ff02ffff04ff07ffff04ffff0bff24ff2480ff8080808080ffff0bff24ff8080808080ffffff02ffff03ffff07ff0580ffff01ff0bffff0102ffff02ff26ffff04ff02ffff04ff09ff80808080ffff02ff26ffff04ff02ffff04ff0dff8080808080ffff01ff0bffff0101ff058080ff0180ff02ff5effff04ff02ffff04ff05ffff04ff0bffff04ffff02ff3affff04ff02ffff04ff09ffff04ff17ffff04ffff02ff26ffff04ff02ffff04ff05ff80808080ff808080808080ffff04ff17ffff04ff2fffff04ff5fffff04ff81bfff80808080808080808080ffff04ffff04ff20ffff04ff17ff808080ffff02ff7cffff04ff02ffff04ff05ffff04ffff02ff82017fffff04ffff04ffff04ff17ff2f80ffff04ffff04ff5fff81bf80ffff04ff0bff05808080ff8202ff8080ffff01ff80808080808080ffff02ff2effff04ff02ffff04ff05ffff04ff0bffff04ffff02ffff03ff3bffff01ff02ff22ffff04ff02ffff04ff05ffff04ff17ffff04ff13ffff04ff2bffff04ff5bffff04ff5fff808080808080808080ffff01ff02ffff03ffff09ff15ffff0bff13ff1dff2b8080ffff01ff0bff15ff17ff5f80ffff01ff088080ff018080ff0180ffff04ff17ffff04ff2fffff04ff5fffff04ff81bfffff04ff82017fff8080808080808080808080ff02ffff03ff05ffff011bffff010b80ff0180ff018080ffff04ffff01ffa024e044101e57b3d8c908b8a38ad57848afd29d3eecc439dba45f4412df4954fdffa0ba42277267f060672a51de6bf75ef3b94bcef71f5964421bfd99b28c3c0ef8e8a0eff07522495060c066f66f32acc2a77e3a3e737aca8baea4d1a64ea4cdc13da9ffff04ffff01ff02ffff01ff02ffff01ff02ffff03ff8202ffffff01ff02ff16ffff04ff02ffff04ff05ffff04ff8204bfffff04ff8206bfffff04ff82017fffff04ffff0bffff19ff2fffff18ffff019100ffffffffffffffffffffffffffffffffff8202ff8080ff0bff82017f80ff8080808080808080ffff01ff04ffff04ff08ffff04ff17ffff04ffff02ff1effff04ff02ffff04ff82017fff80808080ff80808080ffff04ffff04ff1cffff04ff5fffff04ff8206bfff80808080ff80808080ff0180ffff04ffff01ffff32ff3d33ff3effff04ffff04ff1cffff04ff0bffff04ff17ff80808080ffff04ffff04ff1cffff04ff05ffff04ff2fff80808080ffff04ffff04ff0affff04ff5fff808080ffff04ffff04ff14ffff04ffff0bff5fffff012480ff808080ff8080808080ff02ffff03ffff07ff0580ffff01ff0bffff0102ffff02ff1effff04ff02ffff04ff09ff80808080ffff02ff1effff04ff02ffff04ff0dff8080808080ffff01ff0bffff0101ff058080ff0180ff018080ffff04ffff01a032ed0bd89b699a0e70e524037c059ae674d473ccc4c67698df3e2f1fefdd379cffff04ffff01a0a57c0a83eb3a200e28309f095af993bd300bcaa279d2072333e545ea30ae8dfcffff04ffff01b0828357771700a87b388cf9342b72119ca5a55140818f034e3214180fd5306ce39e6c9df1b98e54e924f7c60341cafdc0ffff04ffff01a07319a0d4a23e5c4fbaab34d1ec65b82d00000000000000000000000000000000ffff04ffff01a00e2e3f2e0a5bf43132d9e90695c2ef1089ca93aed226e77663c6f9dfbcc1513fff01808080808080ff01808080',
 'solution': '0xffffa0ee832964616a9655794e40ef43a928788ce713a494e1f45d9e97f68d916aa234ff0180ff01ffff8601977420dc00ff8201e58080'}
2021-06-24T22:09:49.482 wallet chia.wallet.wallet_blockchain: ERROR    Error during db transaction: cannot unpack non-iterable int object
2021-06-24T22:09:49.485 wallet wallet_server              : ERROR    Exception: cannot unpack non-iterable int object, {'host': '127.0.0.1', 'port': 58445}. Traceback (most recent call last):
  File "/root/chia-blockchain/chia/server/server.py", line 523, in wrapped_coroutine
    result = await coroutine
  File "/root/chia-blockchain/chia/wallet/wallet_node_api.py", line 49, in new_peak_wallet
    await self.wallet_node.new_peak_wallet(peak, peer)
  File "/root/chia-blockchain/chia/wallet/wallet_node.py", line 468, in new_peak_wallet
    await self.complete_blocks(blocks, peer)
  File "/root/chia-blockchain/chia/wallet/wallet_node.py", line 417, in complete_blocks
    hbr, trusted=trusted, additional_coin_spends=additional_coin_spends
  File "/root/chia-blockchain/chia/wallet/wallet_blockchain.py", line 238, in receive_block
    block_record, genesis, fork_point_with_peak, additional_coin_spends
  File "/root/chia-blockchain/chia/wallet/wallet_blockchain.py", line 332, in _reconsider_peak
    additional_coin_spends,
  File "/root/chia-blockchain/chia/wallet/wallet_state_manager.py", line 601, in coins_of_interest_received
    await wallet.apply_state_transitions(additional_coin_spends, height)
  File "/root/chia-blockchain/chia/pools/pool_wallet.py", line 293, in apply_state_transitions
    latest_state: Optional[PoolState] = solution_to_extra_data(added_spend)
  File "/root/chia-blockchain/chia/pools/pool_puzzles.py", line 387, in solution_to_extra_data
    return pool_state_from_extra_data(extra_data)
  File "/root/chia-blockchain/chia/pools/pool_puzzles.py", line 357, in pool_state_from_extra_data
    for key, value in extra_data.as_python():
TypeError: cannot unpack non-iterable int object

2021-06-24T22:09:49.485 wallet wallet_server              : ERROR    Exception: cannot unpack non-iterable int object <class 'TypeError'>, closing connection {'host': '127.0.0.1', 'port': 58445}. Traceback (most recent call last):
  File "/root/chia-blockchain/chia/server/server.py", line 533, in api_call
    response: Optional[Message] = await asyncio.wait_for(wrapped_coroutine(), timeout=timeout)
  File "/usr/lib/python3.7/asyncio/tasks.py", line 388, in wait_for
    return await fut
  File "/root/chia-blockchain/chia/server/server.py", line 530, in wrapped_coroutine
    raise e
  File "/root/chia-blockchain/chia/server/server.py", line 523, in wrapped_coroutine
    result = await coroutine
  File "/root/chia-blockchain/chia/wallet/wallet_node_api.py", line 49, in new_peak_wallet
    await self.wallet_node.new_peak_wallet(peak, peer)
  File "/root/chia-blockchain/chia/wallet/wallet_node.py", line 468, in new_peak_wallet
    await self.complete_blocks(blocks, peer)
  File "/root/chia-blockchain/chia/wallet/wallet_node.py", line 417, in complete_blocks
    hbr, trusted=trusted, additional_coin_spends=additional_coin_spends
  File "/root/chia-blockchain/chia/wallet/wallet_blockchain.py", line 238, in receive_block
    block_record, genesis, fork_point_with_peak, additional_coin_spends
  File "/root/chia-blockchain/chia/wallet/wallet_blockchain.py", line 332, in _reconsider_peak
    additional_coin_spends,
  File "/root/chia-blockchain/chia/wallet/wallet_state_manager.py", line 601, in coins_of_interest_received
    await wallet.apply_state_transitions(additional_coin_spends, height)
  File "/root/chia-blockchain/chia/pools/pool_wallet.py", line 293, in apply_state_transitions
    latest_state: Optional[PoolState] = solution_to_extra_data(added_spend)
  File "/root/chia-blockchain/chia/pools/pool_puzzles.py", line 387, in solution_to_extra_data
    return pool_state_from_extra_data(extra_data)
  File "/root/chia-blockchain/chia/pools/pool_puzzles.py", line 357, in pool_state_from_extra_data
    for key, value in extra_data.as_python():
TypeError: cannot unpack non-iterable int object
```

**To Reproduce**

1. checkout pools.testnet9 branch
2. chia init, setup node
3. import keys
4. wait for wallet crash

**Expected behavior**
expected wallet to sync

**Desktop (please complete the following information):**

- OS: Linux
- OS Ubuntu running in docker

**Comment by nextsux** [comment]
wallet stopped syncing at height 522

```
# chia wallet show
Wallet height: 522
Sync status: Not synced
Balances, fingerprint: 958995560
Wallet ID 1 type STANDARD_WALLET Chia Wallet
   -Total Balance: 2.749999999999 txch (2749999999999 mojo)
   -Pending Total Balance: 2.749999999999 txch (2749999999999 mojo)
   -Spendable: 2.749999999999 txch (2749999999999 mojo)
Wallet ID 2 type POOLING_WALLET Pool wallet
   -Total Balance: 0.0 txch (0 mojo)
   -Pending Total Balance: 0.0 txch (0 mojo)
   -Spendable: 0.0 txch (0 mojo)
```

looks like some reorg happened and wallet code can't handle it

**Comment by nextsux** [comment]
probably fixed between 9812442 and 195b4cf works fine now

**Comment by mariano54** [comment]
Fixed, yes
